pkg/systemd/overview-cards/usageCard.jsx

```diff
@@ -29,59 +29,91 @@ import "./usageCard.less";
+            memTotal: Number((info.memory / (1024 * 1024 * 1024)).toFixed(1)),
+            numCpu: info.cpus,
+        }));
+
+        this.onVisibilityChange = this.onVisibilityChange.bind(this);
+        this.onMetricsUpdate = this.onMetricsUpdate.bind(this);
 
-    componentWillUnmount() {
-        this.setState({ pollingEnabled: false });
+        cockpit.addEventListener("visibilitychange", this.onVisibilityChange);
+        this.onVisibilityChange();
     }
 
-    updateMemoryInfo() {
-        if (!this.state.pollingEnabled)
+    onVisibilityChange() {
+        if (cockpit.hidden && this.metrics_channel !== null) {
+            this.metrics_channel.removeEventListener("message", this.onMetricsUpdate);
+            this.metrics_channel.close();
+            this.metrics_channel = null;
             return;
+        }
 
-        machine_info.cpu_ram_info().done(info => {
-            this.setState({
-                memTotal: Number((info.memory / (1024 * 1024 * 1024)).toFixed(1)),
-                memAvailable: Number((info.available_memory / (1024 * 1024 * 1024)).toFixed(1))
-            });
-        });
-        window.setTimeout(this.updateMemoryInfo.bind(this), UPDATE_DELAY);
+        if (!cockpit.hidden && this.metrics_channel === null) {
+            this.metrics_channel = cockpit.channel(METRICS_SPEC);
+            this.metrics_channel.addEventListener("closed", (ev, error) => console.error("metrics closed:", error));
+            this.metrics_channel.addEventListener("message", this.onMetricsUpdate);
+        }
+    }
+
+    onMetricsUpdate(event, message) {
+        const data = JSON.parse(message);
+        // ignore meta messages
+        if (!Array.isArray(data))
+            return;
+        // CPU metrics are in ms/s; divide by 10 to get percentage
+        const cpu = Math.round((data[0][0] + data[0][1] + data[0][2]) / 10 / this.state.numCpu);
+        this.setState({ cpuUsed: cpu, memUsed: Number((data[0][3] / (1024 * 1024 * 1024)).toFixed(1)) });
     }
 
     render() {
-        const memUsed = Number((this.state.memTotal - this.state.memAvailable).toFixed(1));
-        const fraction = memUsed / this.state.memTotal;
+        const fraction = this.state.memUsed / this.state.memTotal;
 
         return (
             <Card className="system-usage">
                 <CardHeader>{_("Usage")}</CardHeader>
                 <CardBody>
                     <table className="pf-c-table pf-m-grid-md pf-m-compact">
                         <tbody>
+                            <tr>
+                                <th scope="row">{_("CPU")}</th>
+                                <td>
+                                    <Progress value={this.state.cpuUsed}
+                                        className="pf-m-sm"
+                                        min={0} max={100}
+                                        variant={this.state.cpuUsed > 90 ? ProgressVariant.danger : ProgressVariant.info}
```

Comment:
I am not sure wether > 90% cpu should count as dangerous.  Sometimes that's exactly what you expect, sometimes it's a sign of a run away process.

Reply:
I managed to see "NaN%" here, but I am not sure how.

Reply:
Should we show the number of cores here?  The point of showing that number is to explain what 100% means, moe than actually telling the user how many cores they have.  In other words, 100% might either mean one full core, or all your cores, and I don't think it's immediately clear which it is.  But if we say "100% of 2 cores", then it is clear.

Reply:
Yes, I think that's a good idea. I added this, taking the same string as in pkg/systemd/graphs.js (to avoid more translation work).